src/sentry/utils/snuba.py

```diff
@@ -296,6 +297,9 @@ def get_snuba_column_name(name):
     if not name or QUOTED_LITERAL_RE.match(name):
         return name
 
+    if TAGKEY_RE.match(name):
```

Comment:
I don't totally follow why we need this. How does a column in the format `tags[hello]` end up getting passed here?

Reply:
It happens when the search/index endpoints are used. They call `snuba.transform_aliases_and_query` which runs all the columns through `get_snuba_column_name` causing the double translation. I'll see if that function can be used here as well. Not using `raw_query` in endpoints would be an improvement.